src/modules/syncedVideoPlayer/SyncedVideoPlayer.ts

```diff
@@ -0,0 +1,78 @@
+import { Socket } from "socket.io-client";
+
+interface User {
+  id: string;
+  name: string;
+}
+
+class SyncedVideoPlayer {
+
+  /**
+   * User {id, name}
+   */
+  user: User
+
+  /**
+   * videoId - youtube videoId
+   */
+  videoId: String
+
+  /**
+   * channel - communication channel
+   */
+  channel: Socket
+
+  /**
+   * room - connected party socket room name
+   * Note: room and channel are the same thing in socket.io
+   */
+  room: String
+
+  /**
+   * player - Youtube player object
+   */
+  player: any
+
+  constructor(user: User, videoId: String, channel: Socket, room: String){
+    this.user = user;
+    this.videoId = videoId;
+    this.room = room;
+    this.channel = channel;
+
+    const id = 'some_id';
+    this.player = new window.YT.Player(`player-${id}`, {
+      videoId: this.videoId,
+      events: {
+        onReady: this.onPlayerReady,
+      },
+    });
+  }
+
+  onPlayerReady() {
+    // TODO
+  }
```

Comment:
This a major part of the `start` implementation please complete it

Reply:
I am a bit confused here, how is this related to `start()`. It's related to the player object which is getting created in the constructor. 
Would you be able to explain what should I implement in it? 

As per my understanding, the player object along with this function should be on the react side utilizing the methods defined under the SyncedVideoPlayer module.

Reply:
No, this is the module which keeps the video in sync